### PR TITLE
rewrite syscall/sysret section

### PR DESCRIPTION
**Pull request by Moldytzu**
Adds some tables with example values taken from my OS. Also explains more in-depth the privilege switching that occurs.

### CODE REVIEW

File: 06_Userspace/04_System_Calls.md [hunk 1]
```diff
@@ -123,13 +123,35 @@ In summary: if the kernel is on a 32-bit platform, use `sysenter`/`sysexit`, for
...
+`STAR` has three fields used to set the GDT descriptors of userspace and kernelspace. The low 32 bits are reserved for the handler of 32-bit variant of `syscall`. It's safe to set it to zero if you don't target 64-bit code. The next 16 bits (47:32) hold the kernel CS/SS base. When `syscall` is executed, the new CS is set to `base` and the new SS is set to `base + 8`. The last 16 bits (64:48) hold the userspace CS/SS base. Because it supports jumping back to compatibility mode, `sysret` requires a very specific GDT layout. It expects the 32-bit CS to be at `base`, the SS (either 64-bit or 32-bit) at `base + 8` and the 64-bit CS to be at `base + 16`. If you don't target 32-bit code, it's safe to omit the 32-bit segment from the GDT, but not that the 64-bit CS will still be fetched from `base + 16`.
```
Comment (DeanoBurrito) [inline 06_Userspace/04_System_Calls.md:130]: > used to set GDT descriptors of

maybe "that specify the selectors used for", a bit more accurate.

> if you dont target 64-bit code

I think this is backwards, you mean its safe to set to zero if you dont support a 32-bit userspace?

> but not that the 64-bit CS

I think that should be "but note that the 64-bit CS"

File: 06_Userspace/04_System_Calls.md [hunk 2]
```diff
@@ -123,13 +123,35 @@ In summary: if the kernel is on a 32-bit platform, use `sysenter`/`sysexit`, for
...
 
-If support compatibility mode is not supported, we can simply omit the 32-bit user code selector, and set use the offset 8 bytes below the user SS. This will work as long as we never try to `sysret` back to compatibility mode.
+`LSTAR` holds the RIP of the handler.
+
+`FMASK` holds a bit mask used to disable certain flags when transitioning control to kernel. Basically, if a bit in `FMASK` is set, the coresponding flag is guranteed to be disabled when entering the kernel.
```
Comment (DeanoBurrito) [inline 06_Userspace/04_System_Calls.md:134]: spelling corrections: coresponding -> corresponding, guranteed -> guaranteed.
Maybe 'disabled' is the wrong word here, perhaps say the flag is cleared?

File: 06_Userspace/04_System_Calls.md [hunk 3]
```diff
@@ -141,23 +163,13 @@ sysretq
...
-
-Finally, we need to tell the CPU we support these instructions and have done all of the above setup. Like many extended features on `x86`, there is a flag to enable them at a global level. For `syscall`/`sysret` this is the system call extensions flag in the EFER MSR, which is bit 0. After setting this, the CPU is ready to handle these instructions!
+Finally, we need to tell the CPU we support these instructions and have done all of the above setup. Like many extended features on `x86`, there is a flag to enable them at a global level. For `syscall`/`sysret` this is the system call extensions flag in the `IA32_EFER` (0xC0000080) MSR, which is bit 0. After setting this, the CPU is ready to handle these instructions!
```
Comment (DeanoBurrito) [inline 06_Userspace/04_System_Calls.md:166]: minor gripe, but `IA32_EFER` is intel's terminology, I think just `EFER` is clear enough.

**Review by DeanoBurrito — CHANGES_REQUESTED** [review]
Thanks for the PR! Mostly looks good, nice writing - couple of minor things and then we'll get it merged.
Also feel free to add yourself to the contributors file at `99_Appendices/I_Acknowledgements.md`

**Comment by Moldytzu** [comment]
This might be it :) Thanks for the suggestions.

**Comment by Moldytzu** [comment]
oops, I missed an important detail after the tables

**Comment by DeanoBurrito** [comment]
lgtm, merged.

**Pull request merged**